pravega/pravega
# Issue 4793: Tablesegment Update APIs.

**Change log description**  
* Extends the logic of `removeTableKeys` and `updateTableEntries` to allow for conditional (conditioned on a matching TableSegment offset) appends.

**Purpose of the change**  
* Closes #4793 
* Alongside PR: #4867, this should conclude #4677.

**What the code does**  
* Adds table segment offset checks in `PravegaRequestProcessor` to both it's `removeTableKeys` and `updateTableEntries` methods. These checks allow these operations to conditonally complete based on whether the provided `tableSegmentOffset` matches the expected offset.
* It first parses the `tableSegmentOffset` from the request(s), compares its value and controls program flow based on this comparison.

**How to verify it**  
(Optional: steps to verify that the changes are effective)

* Run the `PravegaRequestProcessorTests`:
    * This PR only makes changes at the `PravegaRequestProcessor` level. Therefore it is sufficient to only test whether the new code paths are executed given the appropriate conditions.
